shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -159,12 +159,12 @@ abstract class JSBackend(allowUnresolvedSymbols: Bool) {
   protected def translateApp(term: App)(implicit scope: Scope): JSExpr = term match {
     // Binary expressions
     case App(App(Var(op), Tup((N -> Fld(_, lhs)) :: Nil)), Tup((N -> Fld(_, rhs)) :: Nil))
-        if JSBinary.operators contains op =>
-      JSBinary(op, translateTerm(lhs), translateTerm(rhs))
+        if JSBinary.operators contains toJSOperator(op) =>
+      JSBinary(toJSOperator(op), translateTerm(lhs), translateTerm(rhs))
     // Binary expressions with new-definitions
     case App(Var(op), Tup(N -> Fld(_, lhs) :: N -> Fld(_, rhs) :: Nil))
-        if JSBinary.operators.contains(op) && !translateVarImpl(op, isCallee = true).isRight =>
-      JSBinary(op, translateTerm(lhs), translateTerm(rhs))
+        if JSBinary.operators.contains(toJSOperator(op)) && (!translateVarImpl(op, isCallee = true).isRight || op =/= toJSOperator(op)) =>
```

Comment:
Why did these need to change? Please document the reason for these tests and for `toJSOperator`.

Reply:
I documented the reason and used a clearer function name for it.